File: examples/fraud-graph-traversal/02-load.py
"""Step 02 — Load data/transfers.parquet into ArcFlow.

Run:
    uv run python 02-load.py
"""
from __future__ import annotations

import time

from _load import load


def main() -> None:
    t0 = time.perf_counter()
    db, rows = load(verbose=True)
    elapsed = time.perf_counter() - t0
    print(f"loaded in {elapsed:.2f}s ({len(rows) / elapsed:.0f} transfers/sec)")

    n_accts = db.execute("MATCH (a:Account) RETURN count(a)").get(0, 0)
    n_xfer = db.execute("MATCH (t:Transfer) RETURN count(t)").get(0, 0)
    n_sent = db.execute("MATCH ()-[r:SENT]->() RETURN count(r)").get(0, 0)
    n_to = db.execute("MATCH ()-[r:TO]->() RETURN count(r)").get(0, 0)

    print(
        f"in-graph: accounts={n_accts}, transfers={n_xfer}, "
        f"SENT edges={n_sent}, TO edges={n_to}"
    )
    db.close()
    print("OK")


if __name__ == "__main__":
    main()

File: examples/fraud-graph-traversal/rust/src/bin/02-load.rs
//! Step 02 (Rust) — load the transfer fixture from Parquet.
//!
//! Mirrors `02-load.py`. Reads `../data/transfers.parquet` (generated
//! by `../00-make-sample.py`) and builds the Account/Transfer schema.

use anyhow::Result;
use fraud_graph_traversal::load;
use std::path::PathBuf;

fn main() -> Result<()> {
    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
    path.pop(); // rust/ → cookbook root
    path.push("data");
    path.push("transfers.parquet");

    let (db, transfer_count) = load(&path)?;
    println!("loaded {transfer_count} transfers");

    for (label, q) in [
        ("Accounts",  "MATCH (a:Account)  RETURN count(*) AS n"),
        ("Transfers", "MATCH (t:Transfer) RETURN count(*) AS n"),
    ] {
        let result = db.execute(q)?;
        let n = result.rows.first()
            .and_then(|r| r.get("n"))
            .cloned()
            .unwrap_or_else(|| "0".to_string());
        println!("  {label:<10} {n}");
    }
    Ok(())
}
